controllers/heatcfnapi_controller.go

```diff
@@ -336,7 +337,7 @@ func (r *HeatCfnAPIReconciler) reconcileInit(
 
 		ksSvcObj := keystonev1.NewKeystoneService(ksSvcSpec, instance.Namespace, serviceLabels, time.Duration(10)*time.Second)
 		ctrlResult, err = ksSvcObj.CreateOrPatch(ctx, helper)
-		if err != nil {
+		if err != nil || (ctrlResult != ctrl.Result{}) {
```

Comment:
I thought we'd need this logic kepy after condition mirror but it turned out this method returns non-emptry ctrlResult only when KeystoneService is not found... so skipping the condition mirror makes more sense.